include/at_cmd_interface.h

```diff
@@ -0,0 +1,68 @@
+/**
+ * @file at_cmd_interface.h
+ *
+ * @brief Public APIs for the AT command interface driver.
+ */
+
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#ifndef ZEPHYR_INCLUDE_AT_INTERFACE_H_
+#define ZEPHYR_INCLUDE_AT_INTERFACE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/types.h>
+
+/**
+ * @brief AT command return codes
+ */
+enum at_cmd_return_code {
+	AT_CMD_OK,
+	AT_CMD_ERROR,
+	AT_CMD_CMS,
+	AT_CMD_CME,
+	AT_CMD_NOT_RETURN_CODE,
+};
+
+/**
+ * @typedefs at_cmd_interface_handler_t
+ * @brief Callback API when a message is received from the mode
+ *
+ * @param "char * response" Null terminated string containing the modem message
+ * @param "size_t response_len" Length of string
+ *
+ */
+typedef void (*at_cmd_interface_handler_t)(char *response,
+					   size_t response_len);
+
+/**
+ * @brief Function to send an AT command to the modem
+ *
+ * @param cmd Pointer to null terminated AT command string
+ */
+enum at_cmd_return_code at_cmd_interface_write(const char *const cmd);
```

Comment:
I am not sure I want to hightlight where in the stack this driver is located, maybe change the name to at_interface instead. Cmd is anyway not entierly correct, because it's no just commands, it's also notifications.

Reply:
But there is a problem, there is already an at_interface driver in the bsd stack....